src/Plugin/views/field/CustomEntityField.php

```diff
@@ -310,6 +309,38 @@ protected function getItemValue($value, FieldDefinitionInterface $definition) {
     return $value;
   }
 
+
+  /**
+   * Check if date field should be converted to UTC or not.
+   *
+   * @param $definition
+   * @param $date_value
+   *
+   * @return string $datetime_value
+   */
+  public function convertToUTC($definition, $date_value) {
+    $datetime_format = $definition->getSetting('datetime_type') === DateTimeItem::DATETIME_TYPE_DATE ? DateTimeItemInterface::DATE_STORAGE_FORMAT : DateTimeItemInterface::DATETIME_STORAGE_FORMAT;
+    $default_timezone = \Drupal::config('system.date')->get('timezone.default') ?? date_default_timezone_get();
```

Comment:
Wondering why need to load the configuration for the site wide system default timezone. If users are allowed to, and they change their timezone on their user edit page, then that setting would not be respected. date_default_timezone_get() will return the proper timezone, taking into account user's timezone choice.